explainer-beforematch.md

```diff
@@ -0,0 +1,135 @@
+# The `beforematch` event
+
+### Summary (TL;DR)
+
+The `beforematch` event is fired at render timing in the following situations:
+
+* There is a new find-in-page active match, in which case the event is fired
+  with the matched element as the target.
+* There is a fragment-link navigation, in which case the event is fired with
+  the matched fragment as the target.
+* There is a scroll-to-text navigation, in which case the event is fired with
+  the matched element as the target.
+
+### Motivation
+
+With the evolution of the web, there are always new and interesting ways that
+developers choose to organize the information on their pages. Some of these
+approaches (e.g. the common case of text scrolling), yield themselves naturally to
+user-agent features like find-in-page. This is not an accident, since
+find-in-page was designed with common use-cases in mind.
+
+However, there are other interesting approaches, such as collapsed sections of
+text, do not work well with user-agent features since the page does not get any
+indication that the user initiated a find-in-page request, or fragment or
+scroll-to-text navigation.
+
+The `beforematch` event is a signal that is a step in the direction that allows
+us to leverage information that the user-agent has to assist the page in making
+the user experience great. Specifically, with future additions to
+display-locking functionality, it will be possible to process text for find-in-page
+match in sections that are not visible. In turn, the `beforematch` event will be
+fired on hidden (a.k.a. collapsed) sections, allowing the developer to unhide
+the section. The net effect is that the user is able to use find-in-page to find
+content in a collapsed section -- something that is not currently possible.
+
+Even without display-locking features, `beforematch` is useful signal to the
+page which allows custom styling of the matched element, which is now only
+possible with approximations from scroll positions and intersection
+observations.
+
+### Use Cases
+
+#### Example 1
```

Comment:
Make this the last example. Add a first example that is about fragment navigation, both with anchor and text fragment.

Reply:
I think the fragment navigation and the text fragment thing really use the same example.. since it expands the section in the same way that find in page would, so I just added a couple of more paragraphs to that to explain (and reordered the examples)